python/pyspark/sql/connect/dataframe.py

```diff
@@ -1280,10 +1280,20 @@ def __getitem__(self, item: Union[int, str, Column, List, Tuple]) -> Union[Colum
         if isinstance(item, str):
             # Check for alias
             alias = self._get_alias()
-            if alias is not None:
-                return col(alias)
+            _col = alias if alias is not None else item
+            if _col == "*":
+                return Column(UnresolvedStar(unparsed_target=None))
+            elif _col.endswith(".*"):
+                return Column(UnresolvedStar(unparsed_target=_col))
             else:
-                return col(item)
+                if self._plan is None:
```

Comment:
can this really happen?

Reply:
I don't think so

 add this check just because of `self._plan`'s type

`self._plan: Optional[plan.LogicalPlan]`

Reply:
why the plan is optional? this is not related to this PR and we can address later.

Reply:
don't know why. Let's fix it later